families/supplychain/restapi/openapi.yaml

```diff
@@ -0,0 +1,378 @@
+swagger: '2.0'
+
+info:
+  version: "0.8.0"
+  title: Sawtooth Supplychain REST API
+  description: |
+    _This HTTP pragmatic REST API is built on top of Sawtooth Lake's existing
+    ZMQ/Protobuf infrastructure, simplifying SupplyChain client interactions
+    with the blockchain by exposing endpoints that use common HTTP(S)/JSON
+    standards. The SupplyChain REST API provides a secure means for clients to
+    access SupplyChain objects held in global state in a easy manner.
+
+    The REST API provides retrieval using natural keys. Natural keys are unique
+    keys that identify the objects in the real world.
+    (https://en.wikipedia.org/wiki/Natural_key) This provides abstraction of
+    the underlying implementation and achitectures of the Sawtooth blockchain,
+    clients no longer need to know the storage address encoding or the data
+    storage format of SupplyChain objects to access them._
+
+paths:
+  /agents:
+    get:
+      summary: Fetches the data for the current set of Agents
+      description: >
+        Fetches a paginated list of all Agents. Using the `identifier` or name
+        filter parameter, will narrow the list to any Agents that have an
+        identifier or name respectively containing the specified string.
+
+        Agents are identified by their public keys.
+      parameters:
+        - name: identifier
+          in: query
+          type: string
+          description: A partial public_key to filter Agents by
+        - name: name
```

Comment:
Partial name searches are useful, the description specifies contains not starts with filtering. I have expanded the description to specify case insensitivity filtering. so `?name=john` would get `"John Smith"`, `"Smith, John"`, `"Jimmy Johnson"`, `"Billy John Jasper Johanson"`, etc